stdlib/ballerina-mime/src/main/ballerina/mime/Package.md

```diff
@@ -0,0 +1,147 @@
+## Package Overview
+This package provides functions to transmit and receive messages that have multiple parts, such as an attachment. The communication of such messages follow the MIME (Multipurpose Internet Mail Extensions) specification as specified in the RFC 2045 standard.
+### MIME Specific terms 
+The following terms are MIME specific and are extracted from the MIME specification.
+#### Entity
+This refers to the header fields and the content of a message, or a part of the body in a multipart entity. 
+
+#### Body Part
+This refers to an entity that is inside a multipart entity.
+#### Body
+This is the body of an entity, which can be a body of a message or the body of a multipart entity.
+#### Header Fields
+Content-Type, Content-Transfer-Encoding, Content-ID, Content-Description, and Content-Disposition are some MIME header fields. These headers exist along with the other headers in the `Entity`.
+
+```
+Content-Type: image/jpeg
+Content-Disposition: attachment; filename=genome.jpeg;
+Content-Description: a complete map of the human genome
+```
+### Modify and retrieve the data in an `Entity`
+The package provides functions to set and get an entity body from different kinds of message types.  For example, XML, text, JSON, blob, and body parts. Headers can be modified through functions such as `addHeader()`, `setHeader()`, `removeHeader(), etc.
+
+### Encode and decode the body of an `Entity`
+The content of the messages needs to be encoded and decoded when the `Content-Transfer-Encoding` header is used. Ballerina supports Base64 encoding.
+
+The `base64EncodeString()` and `base64DecodeString` encodes and decodes `string` variables. 
```

Comment:
Shall we remove the encode and decode section? Ballerina does the decoding automatically with multiparts before the content is given to the user and it is not a must to encode. It depends on the environment and requirements.